### PR TITLE
[test] Adding platform.json configuration file unit test

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity <sangitamaity0211@gmail.com>

**- What I did**
In order to allow the SONiC community to check in platform capability file i.e. platform.json
file directly under device folder. We need to add this test to make sure the contents of the this file is compliant with platform capability design specified in [DPB HLD doc](https://github.com/zhenggen-xu/SONiC/blob/DPB-design/doc/dynamic-port-breakout/sonic-dynamic-port-breakout-HLD.md)

**- How I did it**
Added platformJson_checker.py file in Test folder.

### - How to verify it
With the change as-is, build passes, the built image executed on the target platform, the configuration was applied properly.

Snapshot of success Build
```
samaity@43b85cf2e5e7:/sonic$ BLDENV=stretch make -f slave.mk target/debs/stretch/sonic-device-data_1.0-1_all.deb
SONiC Build System
 
Build Configuration
"CONFIGURED_PLATFORM"             : "vs"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "admin123"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : "y"
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : ""
"BLDENV"                          : "stretch"
"VS_PREPARE_MEM"                  : "yes"
"ENABLE_SFLOW"                    : "y"
 ```
```
samaity@43b85cf2e5e7:/sonic$ ls -la target/debs/stretch/sonic-device-data_1.0-1_all.deb*
-rw-r--r-- 1 samaity gsamaity 5806088 Dec  4 01:29 target/debs/stretch/sonic-device-data_1.0-1_all.deb
-rw-r--r-- 1 samaity gsamaity   27553 Dec  4 01:29 target/debs/stretch/sonic-device-data_1.0-1_all.deb.log
```

> Snapshot of target/debs/stretch/sonic-device-data_1.0-1_all.deb.log
```
File ../../../device/dell/x86_64-dellemc_z9264f_c3538-r0/media_settings.json passed validity check
File ../../../device/celestica/x86_64-cel_seastone-r0/platform.json passed validity check
/sonic/src/sonic-device-data/src
```
Created a few bad platform.json files to test. Build got failed for all the bad platform capability file.
**Build hit following failure:**

index value is missing
```
Error: index of Ethernet0 has no value.
File ../../../device/celestica/x86_64-cel_seastone-r0/Seastone-DX010-25-50/platform.json failed validity check
Makefile:6: recipe for target 'test' failed
make[3]: *** [test] Error 255
make[3]: Leaving directory '/sonic/src/sonic-device-data/src'
```
interface name is wrong in this case.

```
Error: Unknown Interface Esthernet8s at top level
File ../../../device/celestica/x86_64-cel_seastone-r0/Seastone-DX010-10-50/platform.json failed validity check
Makefile:6: recipe for target 'test' failed
make[3]: *** [test] Error 255
make[3]: Leaving directory '/sonic/src/sonic-device-data/src'
brakout_modes is not the correct port attribute. syntax Error
```
port Attribute is missing
```
File ../../../device/dell/x86_64-dellemc_z9264f_c3538-r0/media_settings.json passed validity check
Error: brakout_modes is not the correct Port attribute.
File ../../../device/celestica/x86_64-cel_seastone-r0/Seastone-DX010-50-40/platform.json failed validity check
Makefile:6: recipe for target 'test' failed
make[3]: *** [test] Error 255
some of the port Attribute is missing
```
Breakout mode is missing here.
```
File ../../../device/celestica/x86_64-cel_seastone-r0/platform.json passed validity check
Error: index, default_brkout_mode of Ethernet16 is/are missing
File ../../../device/celestica/x86_64-cel_seastone-r0/Seastone-DX010/platform.json failed validity check
Makefile:6: recipe for target 'test' failed
make[3]: *** [test] Error 255
```

### CODE REVIEW

**Comment by jleveque** [comment]
Retest vsimage please

File: src/sonic-device-data/tests/platformJson_checker [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline src/sonic-device-data/tests/platformJson_checker:1]: Suggest renaming this file to `platform_json_checker`

File: src/sonic-device-data/tests/platformJson_checker [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    import re
+    import sys
+    import glob
+    import json
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+try:
+  basestring
+except NameError:
+  basestring = str
```
Comment (jleveque) [inline src/sonic-device-data/tests/platformJson_checker:12]: Inconsistent indenting. Please unify all indents for four spaces.

File: src/sonic-device-data/tests/platformJson_checker (hunk 1 above)
Comment (samaity) [reply src/sonic-device-data/tests/platformJson_checker]: #done

File: src/sonic-device-data/tests/platformJson_checker (hunk 2 above)
Comment (samaity) [reply src/sonic-device-data/tests/platformJson_checker]: **#Modified**

File: src/sonic-device-data/tests/platform_json_checker [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    import re
+    import sys
+    import glob
+    import json
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+try:
+    basestring
+except NameError:
+    basestring = str
```
Comment (jleveque) [inline src/sonic-device-data/tests/platform_json_checker:12]: `basestring` has been removed in Python3. We will be migrating to Python 3 soon. Can you please add a TODO comment to remove this for Python 3 and just change to `str`?

File: src/sonic-device-data/tests/platform_json_checker [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ATTR_LEN = len(PORT_ATTRIBUTES)
+PORT_REG = "Ethernet(\d+)"
+PLATFORM_JSON = '*platform.json'
+INTF_KEY = "interfaces"
+
+def usage():
+    print "Usage: " + sys.argv[0] + " <platform_json_file>"
+    sys.exit(1)
+
+
```
Comment (jleveque) [inline src/sonic-device-data/tests/platform_json_checker:25]: Delete extra blank line

File: src/sonic-device-data/tests/platform_json_checker (hunk 3 above)
Comment (samaity) [reply src/sonic-device-data/tests/platform_json_checker:14]: #done

Comment (samaity) [reply src/sonic-device-data/tests/platform_json_checker:14]: #done

File: src/sonic-device-data/tests/platform_json_checker (hunk 4 above)
Comment (samaity) [reply src/sonic-device-data/tests/platform_json_checker]: #done

**Pull request merged**